components/bl-mapbox-component/src/index.js

```diff
@@ -0,0 +1,224 @@
+      mapRef.current.addControl(
+        new MapboxDirections({
+          accessToken: mapboxgl.accessToken,
+        }),
+        'top-left'
+      );
+    }
+  });
+
+  useEffect(() => {
+    if (mapRef.current && center) {
+
+      mapRef.current.setCenter([center.lng, center.lat]);
+    }
+  }, [center]);
+
+  useEffect(() => {
+    mapRef.current.on('load', () => {
+      mapRef.current.setFog({
+        color           : lowerAtmosphere || defaultValues.LOWER_ATMOSPHERE,
+        'high-color'    : upperAtmosphere || defaultValues.UPPER_ATMOSPHERE,
+        'horizon-blend' : atmosphereThickness || defaultValues.ATMOSPHERE_THICKNESS,
+        'space-color'   : spaceColor || defaultValues.SPACE_COLOR,
+        'star-intensity': starIntensity || defaultValues.STAR_INTENSITY,
+      });
+
+      mapRef.current.on('click', e => {
+        const coordinates = { lat: e.lngLat.lat, lng: e.lngLat.lng };
+        onClick({ coordinates: coordinates });
+      });
+
+      mapRef.current.on('move', () => {
+        const center = mapRef.current.getCenter();
+        const southWest = mapRef.current.getBounds().getSouthWest();
+        const northEast = mapRef.current.getBounds().getNorthEast();
+        onPan({ center: center, southWest: southWest, northEast: northEast });
+      });
+
+      if (searchBar) {
+        mapRef.current.addControl(
+          new MapboxGeocoder({
+            accessToken: mapboxgl.accessToken,
+            mapboxgl   : mapboxgl,
+          })
+        );
+      }
+
+      if (fullScreen) {
+        mapRef.current.addControl(new mapboxgl.FullscreenControl());
+      }
+
+      if (navigation) {
+        mapRef.current.addControl(new mapboxgl.NavigationControl());
+      }
+
+      if (geolocation) {
+        const geolocate = new mapboxgl.GeolocateControl({
+          positionOptions  : {
+            enableHighAccuracy: true,
+          },
+          trackUserLocation: true,
+          showUserHeading  : true,
+        });
+        mapRef.current.addControl(geolocate);
+        geolocate.on('trackuserlocationstart', () => {
+          navigator.geolocation.getCurrentPosition(function success(pos) {
+            const coordinates = { lat: pos.coords.latitude, lng: pos.coords.longitude };
+            onDeterminingGeoposition({ coordinates: coordinates });
+          });
+        });
+      }
+
+      createActions(mapRef, component);
+    });
+  }, []);
+
+  useEffect(() => {
+    if (markers) {
+      if (markers.length) {
+        markers.forEach(markerItem => {
+          const marker = new mapboxgl.Marker({ color: markerItem.color })
+            .setLngLat([markerItem.coordinates.lng, markerItem.coordinates.lat])
+            .addTo(mapRef.current);
+
+          if (markerItem.description) {
+            const popup = new mapboxgl.Popup()
+              .setText(markerItem.description);
+
+            marker.setPopup(popup);
+          }
+
+          marker.getElement().addEventListener('click', () => {
```

Comment:
I assume there are created a lot of duplicated handlers after every `markers` change